For the font in the URL bar. Can try to send upstream.

diff --git a/gfx/cairo/cairo/src/cairo-xlib-surface.c b/gfx/cairo/cairo/src/cairo-xlib-surface.c
--- a/gfx/cairo/cairo/src/cairo-xlib-surface.c
+++ b/gfx/cairo/cairo/src/cairo-xlib-surface.c
@@ -1310,7 +1310,11 @@
 		int a, r, g, b;
 
 		if (image_masks.bpp == 1)
+#ifdef WORDS_BIGENDIAN
+		    in_pixel = !! (((uint8_t*)row)[x/8] & (1 << (7 - (x & 7))));
+#else
		    in_pixel = !! (((uint8_t*)row)[x/8] & (1 << (x & 7)));
+#endif
 		else if (image_masks.bpp <= 8)
 		    in_pixel = ((uint8_t*)row)[x];
 		else if (image_masks.bpp <= 16)
